reporter/html.js

```diff
@@ -329,44 +353,106 @@ function toolbarLooseFilter() {
 	return filter;
 }
 
-function toolbarModuleFilterHtml() {
+function moduleDropDownHtml () {
 	var i,
-		moduleFilterHtml = "";
+		dropDownHtml = "";
 
-	if ( !modulesList.length ) {
+	if ( !config.modules.length ) {
 		return false;
 	}
 
-	moduleFilterHtml += "<label for='qunit-modulefilter'>Module: </label>" +
-		"<select id='qunit-modulefilter' name='modulefilter'><option value='' " +
-		( QUnit.urlParams.module === undefined ? "selected='selected'" : "" ) +
-		">< All Modules ></option>";
-
-	for ( i = 0; i < modulesList.length; i++ ) {
-		moduleFilterHtml += "<option value='" +
-			escapeText( encodeURIComponent( modulesList[ i ] ) ) + "' " +
-			( QUnit.urlParams.module === modulesList[ i ] ? "selected='selected'" : "" ) +
-			">" + escapeText( modulesList[ i ] ) + "</option>";
+	for ( i = 0; i < config.modules.length; i++ ) {
+		if ( config.modules[ i ].name !== "" ) {
+			dropDownHtml += "<li><input id='" + config.modules[ i ].moduleId +
+			"' type='checkbox'" +
+			( config.moduleId.indexOf( config.modules[ i ].moduleId ) > -1 ? " checked" : "" ) +
+			"><label for='" + config.modules[ i ].moduleId + "'>" +
+			escapeText( config.modules[ i ].name ) + "</label></li>";
+		}
 	}
-	moduleFilterHtml += "</select>";
-
-	return moduleFilterHtml;
+	return dropDownHtml;
 }
 
-function toolbarModuleFilter() {
-	var toolbar = id( "qunit-testrunner-toolbar" ),
+function toolbarModuleFilter () {
+	var dropDownHtml = moduleDropDownHtml(),
+		toolbar = id( "qunit-testrunner-toolbar" ),
+		label = document.createElement( "label" ),
 		moduleFilter = document.createElement( "span" ),
-		moduleFilterHtml = toolbarModuleFilterHtml();
+		moduleSearch = document.createElement( "input" ),
+		dropDownContainer = document.createElement( "div" ),
+		dropDownList = document.createElement( "ul" ),
+		clearFilter = document.createElement( "li" );
 
-	if ( !toolbar || !moduleFilterHtml ) {
+	if ( !toolbar || !dropDownHtml ) {
 		return false;
 	}
 
-	moduleFilter.setAttribute( "id", "qunit-modulefilter-container" );
-	moduleFilter.innerHTML = moduleFilterHtml;
+	moduleSearch.placeholder = "Select modules";
 
-	addEvent( moduleFilter.lastChild, "change", applyUrlParams );
+	label.innerHTML = "Module: ";
+
+	clearFilter.innerHTML = "< All modules ><hr/>";
```

Comment:
Better yet, drop the `<`, `>`, and `<hr/>` altogether. They seem to only be there for styling, which should be done via CSS instead.